R/admiral_options.R

```diff
@@ -0,0 +1,194 @@
+admiral_default <- new.env(parent = emptyenv())
+admiral_default$admiral_options <- list(
+  subject_keys = vars(STUDYID, USUBJID)
+  # future_input = vars(...) nolint
+
+  ######################################################
+  ### To enhance feature and add inputs as necessary ###
+  ######################################################
+
+  # 1. Add additional options such as future_input as shown commented above
+  # 2. Update @params with future_input in set_admiral_options roxygen documentation
+  # 3. Modify the commented sections of set_admiral_options()
+)
+
+#' Call a package-standard input
+#'
+#' Call a package-standard input that can be modified for advanced users.
+#'
+#' @param input An unquoted expression of commonly used admiral function inputs.
+#'
+#'   As of now, support only available for `subject_keys`.
+#'
+#' @details
+#' This function allows flexibility for function inputs that may need to be repeated
+#' multiple times in a script, such as for `subject_keys`.
+#'
+#' @author Zelos Zhu
+#'
+#' @return
+#' Currently `subject_keys` returns `vars(STUDYID, USUBJID)` unless user changes the default
+#' using `set_admiral_options()`.
+#'
+#' @keywords source_specifications
+#' @family source_specifications
+#'
+#' @export
+#'
+#' @seealso [vars()], [set_admiral_options()], [derive_param_exist_flag()],
+#' [derive_param_first_event()], [derive_param_tte()], [derive_var_disposition_status()],
+#' [derive_var_dthcaus()], [derive_var_extreme_dtm()], [derive_vars_disposition_reason()]
+#'
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate)
+#'
+#' # Derive a new parameter for measurable disease at baseline
+#' adsl <- tibble::tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3"
+#' ) %>%
+#'   mutate(STUDYID = "XX1234")
+#'
+#' tu <- tibble::tribble(
+#'   ~USUBJID, ~VISIT,      ~TUSTRESC,
+#'   "1",      "SCREENING", "TARGET",
+#'   "1",      "WEEK 1",    "TARGET",
+#'   "1",      "WEEK 5",    "TARGET",
+#'   "1",      "WEEK 9",    "NON-TARGET",
+#'   "2",      "SCREENING", "NON-TARGET",
+#'   "2",      "SCREENING", "NON-TARGET"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "XX1234",
+#'     TUTESTCD = "TUMIDENT"
+#'   )
+#'
+#' derive_param_exist_flag(
+#'   dataset_adsl = adsl,
+#'   dataset_add = tu,
+#'   filter_add = TUTESTCD == "TUMIDENT" & VISIT == "SCREENING",
+#'   condition = TUSTRESC == "TARGET",
+#'   false_value = "N",
+#'   missing_value = "N",
+#'   set_values_to = vars(
+#'     PARAMCD = "MDIS",
+#'     PARAM = "Measurable Disease at Baseline"
+#'   ),
+#'   subject_keys = get_admiral_options(subject_keys)
```

Comment:
Yes, for most users these functions work in the background and they won't need to use them and won't need to read the documentation.

I think the documentation is for users who want to change `subject_keys`. In this case I would use `get_admiral_option()` in all places where I need the subject identifiers, e.g., in the example code I stated above. Otherwise, the programs become inconsistent.